### PR TITLE
 gcp - add billing service initial setup

### PR DESCRIPTION
**Pull request by egor-miasnikov**
Added initial classes with `query` and `get` functionality for BillingAccount resource in Billing service

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/billing.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo
+
+
+@resources.register('billingaccount')
```
Comment (bclements) [inline tools/c7n_gcp/c7n_gcp/resources/billing.py]: Change this to cloudbilling-account

File: tools/c7n_gcp/tests/test_billing.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    def test_billingaccount_query(self):
+        session_factory = self.replay_flight_data(
+            'billingaccount-query')
+
+        policy = self.load_policy(
+            self.dryrun_policy,
+            session_factory=session_factory)
+
+        billingaccount_resources = policy.run()
+        self.assertEqual(len(billingaccount_resources), 1)
```
Comment (bclements) [inline tools/c7n_gcp/tests/test_billing.py]: Same commment here about being explicit. If you expect 1 result, ensure you are getting the result you want back. So if you created a billing account called "Foobilling" assert that the policy result returns 1 result named "Foobilling".

**Comment by alexkarpitski** [comment]
@egor-miasnikov @bclements the suggested changes have been done.

File: tools/c7n_gcp/c7n_gcp/resources/billing.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo
+
+
+@resources.register('cloudbilling-account')
+class BillingAccount(QueryResourceManager):
```
Comment (bclements) [inline tools/c7n_gcp/c7n_gcp/resources/billing.py]: Change this to CloudBillingAccount

File: tools/c7n_gcp/tests/test_billing.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gcp_common import BaseTest
+
+
+class BillingAccountTest(BaseTest):
```
Comment (bclements) [inline tools/c7n_gcp/tests/test_billing.py]: Change this to CloudBillingAccountTest

**Comment by alexkarpitski** [comment]
@bclements, renamed the classes as suggested.

File: tools/c7n_gcp/c7n_gcp/resources/billing.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2018-2019 Capital One Services, LLC
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/resources/billing.py]: change to just 2019

File: tools/c7n_gcp/tests/test_billing.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# Copyright 2018-2019 Capital One Services, LLC
```
Comment (jondavidi) [inline tools/c7n_gcp/tests/test_billing.py]: just 2019

**Comment by alexkarpitski** [comment]
@jondavidi, thank you for pointing copyright messages out, I have fixed them as well.

**Comment by egor-miasnikov** [comment]
@bclements @jondavidi is this PR ready for going to `upstream`? If so please click `Approve`

File: tools/c7n_gcp/c7n_gcp/entry.py [hunk 7]
```diff
@@ -16,6 +16,7 @@
 
 import c7n_gcp.policy
 import c7n_gcp.resources.bigquery
+import c7n_gcp.resources.billing
```
Comment (jondavidi) [inline tools/c7n_gcp/c7n_gcp/entry.py]: module needs to be changed to `cloudbilling` as well please

Comment (alexkarpitski) [reply tools/c7n_gcp/c7n_gcp/entry.py]: Sure, done.

**Pull request closed without merging**